src/main/java/de/dhbw/mh/slang/craft/parser/CraftedSlangParser.java

```diff
@@ -332,29 +335,83 @@ AstNode signedTerm3( ){
 	/*===========================================================
 	 * exponentiation
 	 *===========================================================*/
-	
+
+	// Initial Rules:
+	// Exponentiation: atomicExpression "**" exponentiation
+	// Exponentiation: eps
+
+	// There is left-recursion, as the atomicExpression circles over many rules again to Exponentiation
+	// This is noticeable in the select set
+
+	// Rules without left recursion
+	// Exponentiation: atomicExpression Exponent
+	// Exponent: "**" exponentiation
+	// Exponent: eps
+
+
 	@Override
 	public AstNode exponentiation( ){
-		// TODO Auto-generated method stub
-		return super.exponentiation( );
+		// Rule: Exponentiation: atomicExpression Exponent
+		// Select Set: LPAREN, IDENTIFIER, NUMERIC_LITERAL, EOF
+
+		if (LPAREN == LEXER.lookahead().TYPE
+				|| IDENTIFIER == LEXER.lookahead().TYPE
+				|| NUMERIC_LITERAL == LEXER.lookahead().TYPE
+				|| EOF == LEXER.lookahead().TYPE) {
```

Comment:
Why did you include EOF here?

Reply:
When EOF is not included, one tests does not pass. This is why we added it to the  check.
![image](https://github.com/university-lectures/slang/assets/39279223/f3eb26b6-739a-48fe-8a22-fbc410f6ccbc)